Changes for jira PUBSTWO-1646 (rest call for creating pub from sipp)

orosz-usgs opened the pull request:
New mppublications/sipp endpoint in MpPublicationMvcService
  New pojo for json posted: SippMpPubRequest
  Refactored method ISippProcess.processInformationProduct to return MpPublication
  Refactored DisseminationListService.processDisseminationList to have th try/catch inside the for loop.
  Redid tests.
PUBSTWO-1661:
 Fixed typo in ORCID_PREFIX

Commit:
Changes for jira PUBSTWO-1646 (rest call for creating pub from sipp)
  New mppublications/sipp endpoint in MpPublicationMvcService
  New pojo for json posted: SippMpPubRequest
  Refactored method ISippProcess.processInformationProduct to return MpPublication
  Refactored DisseminationListService.processDisseminationList to have th try/catch inside the for loop.
  Redid tests.
PUBSTWO-1661:
 Fixed typo in ORCID_PREFIX

Commit:
Changes for jira PUBSTWO-1646 (rest call for creating pub from sipp)
 Added class SippMpPubRequest

Commit:
Changes for jira PUBSTWO-1646 (rest call for creating pub from sipp)
 Cleanup of MpPublicationMvcService
 Added back Authorization on endpoint

dsteinich commented on src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java:
```diff
@@ -49,24 +48,30 @@ public SippProcess(
 		InformationProduct informationProduct = getInformationProduct(ipNumber);
-		MpPublication mpPublication = getMpPublication(informationProduct);
+
+		if(informationProduct != null) {
```
While unexpected, this check is ok.

dsteinich commented on src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java:
```diff
@@ -49,24 +48,30 @@ public SippProcess(
+		}
+
+		if(mpPublication == null) {
+			mpPublication = new MpPublication();
```
However, this instantiation of an MpPublication is not - code in okToProcess (line 173) expects a null mpPublication if none is found on line 58. Either move this if into the else block below (line 68-73), or remove it and instantiate the MpPublication in line 71 (rtn = ...)

dsteinich commented on src/test/java/gov/usgs/cida/pubs/webservice/mp/MpPublicationMvcServiceTest.java:
```diff
@@ -97,7 +100,7 @@
 	@Before
 	public void setup() {
-		mvcService = new MpPublicationMvcService(pubBusService, busService);
+		mvcService = new MpPublicationMvcService(pubBusService, busService, sippService);
```
I'd like to see some testing of the new end point in here. At the least, to show that the mock service gets called and to see that the posted parameters are validated.

dsteinich commented on src/main/java/gov/usgs/cida/pubs/springinit/SecurityConfig.java:
```diff
@@ -47,23 +47,23 @@ public void configure(ResourceServerSecurityConfigurer resources) {
-					.antMatchers("/mppublications/*/preview*", "/auth/logout").fullyAuthenticated()
-					//authorized
-					.antMatchers("/**").hasAnyAuthority(configurationService.getAuthorizedAuthorities())
+			.and().authorizeRequests().antMatchers("/**").permitAll();
```
We should probably create two different security configs and activate them via spring profiles so we don't accidentally turn of security in production.

orosz-usgs replied:
agreed, the other option would be to figure out Authorization in a dev environment.

Commit:
Changes for jira PUBSTWO-1646 (rest call for creating pub from sipp)
 Updsates per review:
   Added MpPublicationMvcServiceTest.createPubViaSippTest
   Corrected if statement in SippProcess.processInformationProduct
   Reverted SecurityConfig

mbucknell commented on src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippProcess.java:
```diff
@@ -217,35 +223,30 @@ protected ProcessSummary processPublication(ProcessType processType, Information
+		StringBuilder notOkDetails = new StringBuilder("\n\t").append(notProcessedMess).append(" (\"").append(processType).append("\") -")
+				.append(" ProductType: ").append(productType);
+
+		if(informationProduct != null) {
```
I'm surprised Codacy didn't flag this. Should there be a space between **if** and **(**?

Commit:
Changes for jira PUBSTWO-1646 (rest call for creating pub from sipp)
  Format correction to SippProcess flagged in review.
